plugin/pkg/shoot/dns/admission.go

```diff
@@ -193,7 +193,7 @@ func assignDefaultDomainIfNeeded(shoot *garden.Shoot, projectLister gardenlister
 			return apierrors.NewInternalError(err)
 		}
 
-		if shootDomain != nil && strings.HasSuffix(*shootDomain, "."+domain) {
+		if shoot.Spec.DNS != nil && shootDomain != nil && strings.HasSuffix(*shootDomain, "."+domain) {
```

Comment:
Although it doesn't harm, I don't think we need this because it is already checked in line 186.